source/vsm/vsm/db/sqlalchemy/api.py

```diff
@@ -4305,21 +4287,14 @@ def get_poolusage(context, poolusage_id):
     return result
 
 def latency_all_types(context, search_opts, session=None):#for latency
+    # Get diamond collection interval
```

Comment:
That's exactly what I did, I extracted the functions for getting the diamond interval and timestamps into two new functions, at the bottom of the file.  I couldn't refactor it any further because the various metrics functions request timestamps for different metric names, as well as not all using the same return type (some return a list while others return a dict)